shared/src/Twin.cs

```diff
@@ -151,6 +151,19 @@ public Twin(TwinProperties twinProperties)
         [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
         public X509Thumbprint X509Thumbprint { get; internal set; }
 
+        /// <summary>
+        /// The scope of the device. Auto-generated and immutable for edge devices and modifiable in leaf devices to create child/parent relationship.
+        /// </summary>
+        [DefaultValue(null)]
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.IgnoreAndPopulate)]
+        public string DeviceScope { get; internal set; }
```

Comment:
maybe use Scope as the property used in Device to make it clear is same property

Reply:
Thanks for giving it this level of attention, @Ancaantochi. I had thought about that. Here is what I thought about; maybe you can consider that and weigh in again.

I think someone chose to just name it `Device.Scope` because `Device.DeviceScope` sounds repetitive. We also did that for `Device.Id` even though the json property name is `deviceId`.

However, when we get to `Twin.Scope`, I felt it was even more misleading than `Twin.DeviceScope`. I acknowledge it is named differently than in the Device class, but I hoped the comments would help settle that.

I think there is a pro/con to each approach. Given that, what do you think?

Reply:
ok, I think you're right, in Device it uses Scope because it's already a device property, using DeviceScope in twin is more clear.